doc/book/migration/rc-to-v1.md

```diff
@@ -0,0 +1,359 @@
+format as middleware inside those keys, namely:
+
+```php
+[
+    // Required:
+    'middleware' => 'Name of middleware service, or a callable',
+    // Optional:
+    //    'path'  => '/path/to/match',
+    //    'error' => true,
+]
+```
+
+Additionally, the routing and dispatch middleware now become items in the array;
+they (or equivalent entries for your own implementations) must be present in
+your configuration if you want your routed middleware to dispatch!  This change
+gives you full control over the flow of the pipeline.
+
+To specify the routing middleware, use the constant
+`Zend\Expressive\Container\ApplicationFactory::ROUTING_MIDDLEWARE` in place of
+a middleware array; this has the value `EXPRESSIVE_ROUTING_MIDDLEWARE`, if you
+do not want to import the class. Similarly, for the dispatch middleware, use the
+constant `Zend\Expressive\Container\ApplicationFactory::DISPATCH_MIDDLEWARE`
+(value `EXPRESSIVE_DISPATCH_MIDDLEWARE`) to specify the dispatch middleware.
+
+As such, the default configuration now becomes:
+
+```php
+return [
+    'middleware_pipeline' => [
+        // An array of middleware to pipe to the application.
+        // Each item is of the following structure:
+        // [
+        //     // Required:
+        //     'middleware' => 'Name or array of names of middleware services and/or callables',
+        //     // Optional:
+        //     'path'  => '/path/to/match',
+        //     'error' => true,
+        // ],
+        [
+            'middleware' => [
+                Helper\ServerUrlMiddleware::class,
+                Helper\UrlHelperMiddleware::class,
+            ],
+        ],
+
+        // The following is an entry for the routing middleware:
+        Zend\Expressive\Container\ApplicationFactory::ROUTING_MIDDLEWARE,
+
+        // The following is an entry for the dispatch middleware:
+        Zend\Expressive\Container\ApplicationFactory::DISPATCH_MIDDLEWARE,
+
+        // Place error handling middleware after the routing and dispatch
+        // middleware.
+    ],
+]
+```
+
+To update an existing application:
+
+- Promote all `pre_routing` middleware up a level, and remove the `pre_routing`
+  key.
+- Add the entries for `Zend\Expressive\Container\ApplicationFactory::ROUTING_MIDDLEWARE`
+  and `Zend\Expressive\Container\ApplicationFactory::DISPATCH_MIDDLEWARE`
+  immediately following any `pre_routing` middleware, and before any
+  `post_routing` middleware.
+- Promote all `post_routing` middleware up a level, and remove the
+  `post_routing` key.
+
+Once you have made the above changes, you should no longer receive deprecation
+notices when running your application.
+
+## Route result observer deprecation
+
+As of RC6, the following changes have occurred with regards to route result
+observers:
+
+- They are deprecated for usage with `Zend\Expressive\Application`, and that
+  class will not be a route result subject starting in 1.1. You will need to
```

Comment:
Correct, but see my notes above. Technically, we're saying that while the feature will be available in the stable release, *we will not support it*, and have messaged when it will be removed. If we'd already had a stable release, we'd be issuing a new major version; because we haven't, we can mark the feature as deprecated and unsupported.